modules/nickserv/logout.c

```diff
@@ -58,7 +61,27 @@ static void ns_cmd_logout(sourceinfo_t *si, int parc, char *parv[])
 			command_fail(si, fault_nosuch_target, _("\2%s\2 is not logged in."), user);
 			return;
 		}
+	}
+	else if (si->su == NULL)
+	{
+		command_fail(si, fault_needmoreparams, STR_INSUFFICIENT_PARAMS, "LOGOUT");
+		command_fail(si, fault_needmoreparams, _("Syntax: LOGOUT <target> <password>"));
+		return;
+	}
 
+	req.si = si;
+	req.u = u;
+	req.allowed = true;
+	hook_call_user_can_logout(&req);
+	if (!req.allowed)
+	{
+		logcommand(si, CMDLOG_LOGIN, "failed LOGOUT \2%s\2 (denied by hook)", u->nick);
+		command_fail(si, fault_authfail, _("You cannot logout \2%s\2 because the server configuration disallows it."), entity(u->myuser)->name);
```

Comment:
If you want to keep this consistent with the text in nickserv/identify it should be `log out`, not `logout` (since there it is `log in`).